carma_cooperative_perception/src/external_object_list_to_sdsm_component.cpp

```diff
@@ -0,0 +1,167 @@
+// limitations under the License.
+
+#include "carma_cooperative_perception/external_object_list_to_sdsm_component.hpp"
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+
+#include "carma_cooperative_perception/msg_conversion.hpp"
+
+namespace carma_cooperative_perception
+{
+ExternalObjectListToSdsmNode::ExternalObjectListToSdsmNode(const rclcpp::NodeOptions & options)
+: CarmaLifecycleNode{options}
+{
+}
+
+auto ExternalObjectListToSdsmNode::handle_on_configure(
+  const rclcpp_lifecycle::State & /* previous_state */) -> carma_ros2_utils::CallbackReturn
+{
+  RCLCPP_INFO(get_logger(), "Life cycle state transition: configuring");
+
+  sdsm_publisher_ = create_publisher<sdsm_msg_type>("output/sdsms", 1);
+
+  external_objects_subscription_ = create_subscription<external_objects_msg_type>(
+    "input/external_objects", 1, [this](const external_objects_msg_type::SharedPtr msg_ptr) {
+      const auto current_state{this->get_current_state().label()};
+
+      if (current_state == "active") {
+        publish_as_sdsm(*msg_ptr);
+      } else {
+        RCLCPP_WARN(
+          this->get_logger(),
+          "Trying to receive message on the topic '%s', but the containing node is not activated. "
+          "Current node state: '%s'",
+          this->georeference_subscription_->get_topic_name(), current_state.c_str());
+      }
+    });
+
+  georeference_subscription_ = create_subscription<georeference_msg_type>(
+    "input/georeference", 1, [this](const georeference_msg_type::SharedPtr msg_ptr) {
+      const auto current_state{this->get_current_state().label()};
+
+      if (current_state == "active") {
+        update_georeference(*msg_ptr);
+      } else {
+        RCLCPP_WARN(
+          this->get_logger(),
+          "Trying to receive message on the topic '%s', but the containing node is not "
+          "activated. "
+          "Current node state: '%s'",
+          this->georeference_subscription_->get_topic_name(), current_state.c_str());
+      }
+    });
+
+  current_pose_subscription_ = create_subscription<pose_msg_type>(
+    "input/pose_stamped", 1, [this](const pose_msg_type::SharedPtr msg_ptr) {
+      const auto current_state{this->get_current_state().label()};
+
+      if (current_state == "active") {
+        update_current_pose(*msg_ptr);
+      } else {
+        RCLCPP_WARN(
+          this->get_logger(),
+          "Trying to recieve message on topic '%s', but the containing node is not activated."
+          "current node state: '%s'",
+          this->current_pose_subscription_->get_topic_name(), current_state.c_str());
+        
+      }
+    });
+
+  return carma_ros2_utils::CallbackReturn::SUCCESS;
+}
+
+auto ExternalObjectListToSdsmNode::handle_on_activate(
+  const rclcpp_lifecycle::State & /* previous_state */) -> carma_ros2_utils::CallbackReturn
+{
+  RCLCPP_INFO(get_logger(), "Life cycle state transition: activating");
+
+  sdsm_publisher_->on_activate();
```

Comment:
Lifecycle publishers in classes inheriting from `CarmaLifecycleNode` do not need to manage the lifecycle of lifecycle publishers (assuming they have been added to the `lifecycle_publishers_` list.